core/app/models/spree/product/scopes.rb

```diff
@@ -99,31 +99,29 @@ def self.property_conditions(property)
     # note that it can test for properties with NULL values, but not for absent values
     add_search_scope :with_property_value do |property, value|
       joins(:properties)
-        .where("#{ProductProperty.table_name}.value = ?", value)
+        .where(ProductProperty.arel_table[:value].eq(value))
         .where(property_conditions(property))
     end
 
     add_search_scope :with_option do |option|
-      option_types = OptionType.table_name
       conditions = case option
-      when String     then { "#{option_types}.name" => option }
-      when OptionType then { "#{option_types}.id" => option.id }
-      else                 { "#{option_types}.id" => option.to_i }
-      end
+                     when String     then OptionType.arel_table[:name].eq(option)
+                     when OptionType then OptionType.arel_table[:id].eq(option.id)
+                     else                 OptionType.arel_table[:id].eq(option.to_i)
+                   end
 
       joins(:option_types).where(conditions)
     end
 
     add_search_scope :with_option_value do |option, value|
-      option_values = OptionValue.table_name
       option_type_id = case option
-        when String then OptionType.find_by(name: option) || option.to_i
-        when OptionType then option.id
-        else option.to_i
-      end
+                         when String then OptionType.find_by(name: option) || option.to_i
+                         when OptionType then option.id
+                         else option.to_i
+                       end
 
-      conditions = "#{option_values}.name = ? AND #{option_values}.option_type_id = ?", value, option_type_id
-      group('spree_products.id').joins(variants_including_master: :option_values).where(conditions)
+      group(Product.arel_table[:id]).joins(variants_including_master: :option_values)
+        .where(OptionValue.arel_table[:name].eq(value), OptionValue.arel_table[:option_type_id].eq(option_type_id))
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Align `.where` with `.joins` on line 123.